ogr/ogrsf_frmts/miramon/mm_gdal_functions.c

```diff
@@ -0,0 +1,2801 @@
+        (long_nom_camp >= MM_MAX_LON_CLASSICAL_FIELD_NAME_DBF))
+    {
+        cadena[MM_MAX_LON_FIELD_NAME_DBF - 1] = '\0';
+        error_retornat |= MM_FIELD_NAME_TOO_LONG;
+    }
+    error_retornat |= MM_return_common_valid_DBF_field_name_string(cadena);
+    return error_retornat;
+}
+
+static MM_BOOLEAN
+MM_CheckClassicFieldNameEqual(const struct MM_DATA_BASE_XP *data_base_XP,
+                              const char *classical_name)
+{
+    MM_EXT_DBF_N_FIELDS i;
+
+    for (i = 0; i < data_base_XP->nFields; i++)
+    {
+        if ((strcasecmp(data_base_XP->pField[i].ClassicalDBFFieldName,
+                        classical_name)) == 0 ||
+            (strcasecmp(data_base_XP->pField[i].FieldName, classical_name)) ==
+                0)
+            return TRUE;
+    }
+    return FALSE;
+}
+
+static char *MM_GiveNewStringWithCharacterInFront(const char *text,
+                                                  char caracter)
+{
+    char *ptr;
+    size_t i;
+
+    if (!text)
+        return nullptr;
+
+    i = strlen(text);
+    if ((ptr = calloc_function(i + 2)) == nullptr)
+        return nullptr;
+
+    *ptr = caracter;
+    memcpy(ptr + 1, text, i + 1);
+    return ptr;
+}
+
+static char *MM_SetSubIndexFieldNam(char *nom_camp, MM_EXT_DBF_N_FIELDS index,
+                                    size_t ampladamax)
+{
+    char *NomCamp_SubIndex;
+    char *_subindex;
+    char subindex[15];
+    size_t sizet_subindex;
+    size_t sizet_nomcamp;
+
+    NomCamp_SubIndex = calloc_function(ampladamax * sizeof(char));
+    if (!NomCamp_SubIndex)
+        return nullptr;
+
+    strcpy(NomCamp_SubIndex, nom_camp);
+
+    sprintf(subindex, sprintf_UINT64, (GUInt64)index);
+
+    _subindex = MM_GiveNewStringWithCharacterInFront(subindex, '_');
+    sizet_subindex = strlen(_subindex);
+    sizet_nomcamp = strlen(NomCamp_SubIndex);
+
+    if (sizet_nomcamp + sizet_subindex > ampladamax - 1)
+        memcpy(NomCamp_SubIndex + ((ampladamax - 1) - sizet_subindex),
+               _subindex, strlen(_subindex));
+    else
+        NomCamp_SubIndex = strcat(NomCamp_SubIndex, _subindex);
+
+    free_function(_subindex);
+
+    return NomCamp_SubIndex;
+}
+
+MM_FIRST_RECORD_OFFSET_TYPE
+MM_GiveOffsetExtendedFieldName(const struct MM_FIELD *camp)
+{
+    MM_FIRST_RECORD_OFFSET_TYPE offset_nom_camp;
+
+    memcpy(&offset_nom_camp,
+           (char *)(&camp->reserved_2) + MM_OFFSET_RESERVAT2_OFFSET_NOM_ESTES,
+           4);
+    return offset_nom_camp;
+}
+
+int MM_WriteNRecordsMMBD_XPFile(struct MMAdmDatabase *MMAdmDB)
+{
+    GUInt32 nRecords;
+    if (!MMAdmDB->pMMBDXP || !MMAdmDB->pFExtDBF)
+        return 0;
+
+    // Updating number of features in features table
+    fseek_function(MMAdmDB->pFExtDBF, MM_FIRST_OFFSET_to_N_RECORDS, SEEK_SET);
+
+    if (MMAdmDB->pMMBDXP->nRecords > UINT32_MAX)
+    {
+        MMAdmDB->pMMBDXP->dbf_version = MM_MARCA_VERSIO_1_DBF_ESTESA;
+
+        if (fwrite_function(&MMAdmDB->pMMBDXP->nRecords, 4, 1,
```

Comment:
ok, I see you are writing directly integers to a file without taking into account potential big-endian order. Making the driver compliant with a big-endian host is likely to be a significant and painful amount of work, so the best course of action would be to not build the driver on big endian hosts. Otherwise you'll get test errors when the pull request will be done against OSGeo/gdal which has 2 extra CI configuration using Travis-CI, one including IBM s390x which is big-endian
So apply the following change to ogr/ogrsf_frmts/CMakeLists.txt:
```
diff --git a/ogr/ogrsf_frmts/CMakeLists.txt b/ogr/ogrsf_frmts/CMakeLists.txt
index cd608d1f53..f11285935d 100644
--- a/ogr/ogrsf_frmts/CMakeLists.txt
+++ b/ogr/ogrsf_frmts/CMakeLists.txt
@@ -47,7 +47,9 @@ ogr_optional_driver(vdv "VDV-451/VDV-452/INTREST Data Format")
 ogr_optional_driver(flatgeobuf FlatGeobuf)
 ogr_optional_driver(mapml MapML)
 ogr_optional_driver(jsonfg JSONFG)
-ogr_optional_driver(miramon "MiraMonVector")
+if( NOT WORDS_BIGENDIAN )
+    ogr_optional_driver(miramon "MiraMonVector")
+endif()
```

Reply:
It would be clearer that you only write the 32-bit low bits with something like:
GUInt32 nRecords32LowBits = (GUInt32)(MMAdmDB->pMMBDXP->nRecords & UINT32_MAX);
and fwrite_function() that

Reply:
done